app/models/government.rb

```diff
@@ -20,4 +22,53 @@ def self.on_date(date)
   def current?
     self == Government.current
   end
+
+  def overlaps?(other)
+    !before?(other) && !after?(other)
+  end
+
+  def ended?
+    end_date.present?
+  end
+
+private
+
+  def before?(other)
+    starts_before?(other) && ends_before?(other)
+  end
+
+  def after?(other)
+    starts_after?(other) && ends_after?(other)
+  end
+
+  def starts_before?(other)
+    start_date < other.start_date
+  end
+
+  def ends_before?(other)
+    self.ended? && end_date <= other.start_date
+  end
+
+  def starts_after?(other)
+    other.ended? && start_date >= other.end_date
+  end
+
+  def ends_after?(other)
+    return true unless self.ended?
```

Comment:
What if you're comparing an ended government with another that has ended? Will this return true regardless of which ended first?